controllers/runtimecomponent_controller.go

```diff
@@ -146,7 +147,13 @@ func (r *RuntimeComponentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		// it is not possible to override converted annotations.
 		instance.Annotations = appstacksutils.MergeMaps(instance.Annotations, appstacksutils.GetOpenShiftAnnotations(instance))
 	}
-
+	// If TargetPorts on Serviceports are not set, default them to the Port value in the CR instance
```

Comment:
It'll be good to do this at the end of the Initialize() method called above. If the actual logic to go through the ports is added to a common util method, other operators can also invoke it from their Initialize methods.

Reply:
@leochr  I've move the logic into the initialize() method but couldn't move it into the common utils as it would create a circular import